examples/protocols/https_server/main/certs/gencert.sh

```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+openssl req -newkey rsa:2048 -nodes -keyout prvtkey.pem -x509 -days 3650 -out cacert.pem -subj "/CN=ESP32 HTTPS server example"
```

Comment:
Could you instead put this inside README.md

Reply:
I can add it there, but why is this a problem? 
Here you just run the script and it generates a new certificate for the example

Reply:
If you wish to keep the executable

* either you need to remove the executable flag (and thereby add an extra step in README to enable the executable flag)
* or you would have to modify `esp-idf/tools/ci/executable-list.txt` (in which you would find no example related scripts till now)

Not doing either of the above will cause our CI tests to fail.

This is asserted because we wish to keep the number of executable in IDF to a minimum (only those which are essential, like scripts related to build system)

Reply:
I see. I'd favor removing the exacutable flag, but it's ultimately up to you, what matters is to keep the example command.

The OpenSSL server example (https://github.com/espressif/esp-idf/tree/master/examples/protocols/openssl_server), that I used to learn about openSSL, references a document called "ESP8266_SDKSSL_User_Manual_EN_v1.4.pdf". That document doesn't exist, I didn't find anything at all on Google. The linked page has a file called "5a-esp8266_sdk_ssl_user_manual_en.pdf", but I didn't really understand it.

so it's good to keep it for reference. Also it's very important to make generating the certificate easy, or people might blindly reuse the example certificate.

Reply:
> it's good to keep it for reference. Also it's very important to make generating the certificate easy, or people might blindly reuse the example certificate.

I agree. This is a minor thing. What matters is that it should be convenient for the user.

Then you can keep the file with executable flag removed. If every one is okay with this during the internal review then it should be alright.